python-package/lightgbm/basic.py

```diff
@@ -509,7 +507,14 @@ def is_allowed_numpy_dtype(dtype):
             and not issubclass(dtype, (np.timedelta64, float128))
         )
 
-    return [i for i, dtype in enumerate(dtypes) if not is_allowed_numpy_dtype(dtype.type)]
+    bad_pandas_dtypes = [
+        f'{column_name}: {pandas_dtype}'
+        for column_name, pandas_dtype in pandas_dtypes_series.iteritems()
+        if not is_allowed_numpy_dtype(pandas_dtype.type)
+    ]
+    if bad_pandas_dtypes:
+        raise ValueError('DataFrame.dtypes must be int, float or bool.\n'
```

Comment:
Maybe replace concrete `DataFrame` with just `pandas` for more general wording that wouldn't be confusing for the case of label as a `Series`?
https://github.com/microsoft/LightGBM/pull/5065#issuecomment-1064708046
```suggestion
        raise ValueError('pandas dtypes must be int, float or bool.\n'
```

Reply:
^ I agree with this